### PR TITLE
chore: sm update to v1.5.0

### PR DESCRIPTION
**Pull request by kierramarie**
### Description

Update to secrets manager v1.5.0. Rename service_endpoints to allowed_network. Complete example has been updated.

### Release required?

- [x] No release
- [ ] Patch release (`x.x.X`)
- [ ] Minor release (`x.X.x`)
- [ ] Major release (`X.x.x`)

### Run the pipeline

If the CI pipeline doesn't run when you create the PR, the PR requires a user with GitHub collaborators access to run the pipeline.

Run the CI pipeline when the PR is ready for review and you expect tests to pass. Add a comment to the PR with the following text:

```
/run pipeline
```

### Checklist for reviewers

- [ ] If relevant, a test for the change is included or updated with this PR.
- [ ] If relevant, documentation for the change is included or updated with this PR.

### For mergers

- Use a conventional commit message to set the release level. Follow the [guidelines](https://terraform-ibm-modules.github.io/documentation/#/merging.md).
- Include information that users need to know about the PR in the commit message. The commit message becomes part of the GitHub release notes.
- Use the **Squash and merge** option.

### CODE REVIEW

**Comment by ocofaigh** [comment]
```
Error: [ERROR] Error when creating resource instance: Please contact the Service Provider for this error. [422, Unprocessable Entity] This account reached the number of Trial plan instances that are allowed. To proceed, you must select the Standard plan or delete an existing Trial instance. If you delete the Trial plan instance, you must also delete its reclamation by using the IBM Cloud CLI. Learn more: https://ibm.biz/BdPa49 with resp code
```

We hit out quota limit, but should be good now so re-running

**Review by alex-reiff — DISMISSED** [review]

**Review by shemau — CHANGES_REQUESTED** [review]
That does not work.

File: examples/default/version.tf [hunk 1]
```diff
@@ -3,7 +3,7 @@ terraform {
   required_providers {
     ibm = {
       source  = "IBM-Cloud/ibm"
-      version = "= 1.56.1"
+      version = ">= 1.56.1, < 2.0.0"
```
[the remark this answers is not in the document]
Comment (shemau) [reply examples/default/version.tf:6]: @ocofaigh thoughts changing the root module to 1.62.0 (required by SM in example for CBR fix)?

Using an existing SM instance would not work well here, due to nature of engines; 
or stop using the SM module and create a basic example with resource that creates an SM.

Issues with trial licenses and expense etc. Maybe just bump the whole module to IBM provider 1.62.0?

[the remark this answers is not in the document]
Comment (ocofaigh) [reply examples/default/version.tf:6]: The trial quota we are hitting has nothing to do with any code changes here. We just currently have alot of instances being provisioned in parallel due to the amount of paper cut items currently. 

This repo is one of the one where we kind of can't have a basic example that calls no other modules. The engine is going to need an SM instance. I'm thinking that maybe we bump the root level module to match the same as the SM module for this one. Its likely that these modules would be deployed as part of the same state so this is probably OK to do anyway.

[the remark this answers is not in the document]
Comment (kierramarie) [reply examples/default/version.tf:6]: @ocofaigh is there anything else I need to do for this PR?

**Comment by alex-reiff** [comment]
looks like the changes got covered in another PR, I don't see anything here.

**Comment by kierramarie** [comment]
@alex-reiff ah, you're right. Closing this issue.

**Pull request closed without merging**